Composer/packages/lib/bot-deploy/src/botProjectDeploy.ts

```diff
@@ -84,12 +85,55 @@ export class BotProjectDeploy {
     this.generatedFolder = config.generatedFolder ?? path.join(this.remoteBotPath, 'generated');
   }
 
+  private getErrorMesssage(err) {
+    if (err.body) {
+      if (err.body.error) {
+        if (err.body.error.details) {
+          const details = err.body.error.details;
+          let errMsg = '';
+          for (let detail of details) {
+            errMsg += detail.message;
+          }
+          return errMsg;
+        } else {
+          return err.body.error.message;
+        }
+      } else {
+        return JSON.stringify(err.body, null, 2);
+      }
+    } else {
+      return JSON.stringify(err, null, 2);
+    }
+  }
+
   private pack(scope: any) {
     return {
       value: scope,
     };
   }
 
+  /**
+   * For more information about this api, please refer to this doc: https://docs.microsoft.com/en-us/rest/api/resources/Tenants/List
+   */
+  private async getTenantId() {
+    if (!this.accessToken) {
+      throw new Error(
+        'Error: Missing access token. Please provide a non-expired Azure access token. Tokens can be obtained by running az account get-access-token'
+      );
+    }
+    try {
+      const tenantUrl = `https://management.azure.com/tenants?api-version=2020-01-01`;
+      const options = {
+        headers: { Authorization: `Bearer ${this.accessToken}` },
+      } as rp.RequestPromiseOptions;
+      const response = await rp.get(tenantUrl, options);
+      const jsonRes = JSON.parse(response);
+      return jsonRes.value[0].tenantId;
```

Comment:
is it guaranteed to be at least 1 item in this array?

seems like something we should potentially error check

Reply:
The documentation for the api is not super clear what happens when there is no tenants. However this api lists all tenants for the account. Two things stem from this:

- The case where there are zero values in this array is when your account has no tenants. I'm not sure if this is possible, but we can easily just print an error message if this happens and there are zero elements in the json response.
- Accounts with multiple tenants (which is a thing that is allowed) will be forced to use the first tenant in the list. This is a problem that we should fix at some point. We might want to file an issue to track this. Once we allow for azure sign in, the tenant issue will exacerbate, since the tenant selection is important for resource creation and AAD app creation.

Reply:
Has been fixed, add more error handling .